api/src/daily-coding-challenge/routes/daily-coding-challenge.test.ts

```diff
@@ -92,18 +92,29 @@ describe('/daily-coding-challenge', () => {
     });
 
     it('should return 400 for an invalid date format', async () => {
-      const res = await superRequest(
-        '/daily-coding-challenge/date/invalid-format',
-        {
-          method: 'GET'
-        }
-      ).send({});
+      const invalidFormats = [
+        'invalid-format',
+        '2025-07',
+        '07-18-2025',
+        '25-07-18',
+        '2025-7-18',
+        '2025-07-8'
+      ];
```

Comment:
Do not mean to be pedantic but I think it would be better to check for valid format only and reject the rest. I mean negative checks are sometimes source of buggy logic.